modules/031-linstor/images/linstor-pools-importer/main.go

```diff
@@ -0,0 +1,481 @@
+			if err != nil {
+				klog.Fatalln("failed to generate Kubernetes storage class:", err)
+			}
+			err = kc.Get(ctx, types.NamespacedName{Name: storageClass.GetName()}, &storageClass)
+			if err == nil { // TODO check for 404
+				continue
+			}
+			err = kc.Create(ctx, &storageClass)
+			if err != nil {
+				event := makeKubernetesEvent(&owner, v1.EventTypeWarning, "Failed", "Failed to create Kubernetes storage class"+err.Error())
+				if err := kc.Create(ctx, &event); err != nil {
+					klog.Errorln("Failed to create event", err)
+				}
+				klog.Fatalln("Failed to create Kubernetes storageClass", err)
+			}
+			event := makeKubernetesEvent(&owner, v1.EventTypeNormal, "Created", "Created Kubernetes storage class: "+storageClass.Name)
+			err = kc.Create(ctx, &event)
+			if err != nil {
+				klog.Fatalln("Failed to create event", err)
+			}
+		}
+	}
+}
+
+// Makes loop over storage pool candidates, retruns channel of changed ones
+func runCandidatesLoop(ctx context.Context, f func() ([]StoragePoolCandidate, error), delay time.Duration) <-chan StoragePoolCandidate {
+	ch := make(chan StoragePoolCandidate)
+	go func() {
+		var oldCandidates []StoragePoolCandidate
+		for {
+			select {
+			case <-ctx.Done():
+				close(ch)
+				return
+			default:
+				candidates, err := f()
+				if err != nil {
+					klog.Fatalln("failed to load candidates:", err)
+				}
+			LOOP:
+				for _, candidate := range candidates {
+					name := candidate.GetName()
+					if name == "" {
+						klog.Fatalln("storage pool name can't be empty")
+					}
+					for _, oldCandidate := range oldCandidates {
+						if name == oldCandidate.GetName() {
+							continue LOOP
+						}
+					}
+					ch <- candidate
+				}
+				oldCandidates = candidates
+			}
+			time.Sleep(delay)
+		}
+	}()
+	return ch
+}
+
+// Collects all storage pool candidates from the node
+func getCandidates() ([]StoragePoolCandidate, error) {
+	var candidates []StoragePoolCandidate
+
+	// Getting LVM volume groups
+	var vgs VolumeGroups
+	err := vgs.LoadCandidates()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get LVM volume groups: %s", err)
+	}
+	for _, vg := range vgs {
+		candidates = append(candidates, vg)
+	}
+
+	// Getting LVM thin pools
+	var tps ThinPools
+	err = tps.LoadCandidates()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get LVM thin pools: %s", err)
+	}
+	for _, tp := range tps {
+		candidates = append(candidates, tp)
+	}
+
+	return candidates, nil
+}
+
+type StoragePoolCandidates interface {
+	LoadCandidates() error
+}
+
+// Defines any type of LINSTOR storage pool candidate
+type StoragePoolCandidate interface {
+	GetName() string
+	GetProviderKind() lclient.ProviderKind
+	GetProps() map[string]string
+}
+
+type ThinPools []ThinPool
+type ThinPool struct {
+	Name   string
+	VGName string
+	Tags   []string
+}
+
+func (a *ThinPools) LoadCandidates() error {
+	cmd := exec.Command("lvs", "-oname,vg_name,lv_attr,tags", "--separator=;", "--noheadings", "--config="+lvmConfig)
```

Comment:
Because they must implement `StoragePoolCandidates` interface.
I supposed that in the future we can add another drivers eg. ZFS.

This way the driver realization will require just to implement these two interfaces:

https://github.com/deckhouse/deckhouse/blob/041985b8f84e7ac9b4d92448dc0d60bf4281d3bd/modules/031-linstor/images/linstor-pools-importer/main.go#L253-L262

Implementation of `LoadCandidates()` is always driver specific, but output is standardized.